### PR TITLE
add --set-filename option

### PR DESCRIPTION
**Pull request by rps-repo**
added set filename option to _encrypt method.

### CODE REVIEW

**Comment by alecpl** [comment]
If we implement this it should be optional. I would not use it for my use-cases. There should be a test for this case too. Probably could be just a "master user feature" described in https://github.com/pear/Crypt_GPG/pull/31.

File: Crypt/GPG.php [hunk 1]
```diff
@@ -1639,6 +1639,13 @@ protected function _encrypt($data, $isFile, $outputFile, $armor)
             $arguments[] = '--recipient ' . escapeshellarg($key['fingerprint']);
         }
 
+        if ($isFile){
```
Comment (CloCkWeRX) [inline Crypt/GPG.php]: Prefer ```if () {``` formatting

File: Crypt/GPG.php [hunk 2]
```diff
@@ -1639,6 +1639,13 @@ protected function _encrypt($data, $isFile, $outputFile, $armor)
...
+            $pathParts = pathinfo($data);
+            if (!empty($pathParts['basename'])){
```
Comment (CloCkWeRX) [inline Crypt/GPG.php]: Ditto

File: composer.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "pear/crypt_gpg",
+    "name": "farpointsystems/crypt_gpg",
```
Comment (CloCkWeRX) [inline composer.json]: We wouldn't want this one in the main repo

**Comment by rps-repo** [comment]
OK, changes made.

**Comment by alecpl** [comment]
See 40de9f2246d0517.

**Pull request closed without merging**